tests/helpers/test_script.py

```diff
@@ -3538,6 +3538,98 @@ async def test_if_condition_validation(
     )
 
 
+async def test_sequence(hass: HomeAssistant, caplog: pytest.LogCaptureFixture) -> None:
+    """Test sequence action."""
+    events = async_capture_events(hass, "test_event")
+
+    sequence = cv.SCRIPT_SCHEMA(
+        [
+            {
+                "alias": "Sequential group",
+                "sequence": [
+                    {
+                        "alias": "sequence group, action 1",
+                        "event": "test_event",
+                        "event_data": {
+                            "sequence": "group",
+                            "action": "1",
+                            "what": "{{ what }}",
+                        },
+                    },
+                    {
+                        "alias": "sequence group, action 2",
+                        "event": "test_event",
+                        "event_data": {
+                            "sequence": "group",
+                            "action": "2",
+                            "what": "{{ what }}",
+                        },
+                    },
+                ],
+            },
+            {
+                "alias": "action 2",
+                "event": "test_event",
+                "event_data": {"action": "2", "what": "{{ what }}"},
+            },
+        ]
+    )
+
+    script_obj = script.Script(hass, sequence, "Test Name", "test_domain")
+
+    await script_obj.async_run(MappingProxyType({"what": "world"}), Context())
+
+    assert len(events) == 3
+    assert events[0].data["sequence"] == "group"
+    assert events[0].data["action"] == "1"
+    assert events[0].data["what"] == "world"
+    assert events[1].data["sequence"] == "group"
+    assert events[1].data["action"] == "2"
+    assert events[1].data["what"] == "world"
+    assert "sequence" not in events[2].data
+    assert events[2].data["action"] == "2"
+    assert events[2].data["what"] == "world"
```

Comment:
In my opinion the following is more readable and more strict.
```suggestion
    assert events[0].data == {
        "sequence": "group",
        "action": "1",
        "what": "world",
    }
    assert events[1].data == {
        "sequence": "group",
        "action": "2",
        "what": "world",
    }

    assert events[2].data == {
        "action": "2",
        "what": "world",
    }
```

Reply:
Agree